src/Compilers/CSharp/Portable/Symbols/AssemblySymbol.cs

```diff
@@ -442,19 +442,16 @@ internal bool RuntimeSupportsStaticAbstractMembersInInterfaces
 
         /// <summary>
         /// Whether the target runtime supports numeric IntPtr types.
-        /// This test hook should be removed once TargetFramework.Net70 is added.
-        /// Tracked by https://github.com/dotnet/roslyn/issues/61235
         /// </summary>
-        internal virtual bool RuntimeSupportsNumericIntPtr
+        internal bool RuntimeSupportsNumericIntPtr
         {
             get
             {
                 // CorLibrary should never be null, but that invariant is broken in some cases for MissingAssemblySymbol.
                 // Tracked by https://github.com/dotnet/roslyn/issues/61262
-                return CorLibrary?.RuntimeSupportsNumericIntPtr == true;
+                return CorLibrary is not null &&
+                    RuntimeSupportsFeature(SpecialMember.System_Runtime_CompilerServices_RuntimeFeature__NumericIntPtr);
```

Comment:
Consider separating product changes and infrastructure changes into separate PRs